avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -691,9 +691,6 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
       case STRING:
         if (enhancedSchemaSupport && schema.parameters() != null
             && schema.parameters().containsKey(AVRO_TYPE_ENUM)) {
-          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
-          org.apache.avro.Schema cachedEnum = schemaMap.get(enumSchemaName);
-          if (cachedEnum == null) {
```

Comment:
Need someone to verify this change, but i believe if the caching is taking place via the hashCode of the input Schema, then it should only reach the branch of this switch iff it wasn't already cached. My concern is if this has any adverse effect based on the isOptional and ignoreOptional parameters.